compiler/rustc_privacy/src/lib.rs

```diff
@@ -112,13 +113,26 @@ where
                 ty.visit_with(self)
             }
             ty::PredicateKind::RegionOutlives(..) => ControlFlow::CONTINUE,
-            ty::PredicateKind::ConstEvaluatable(..)
+            ty::PredicateKind::ConstEvaluatable(_, substs)
```

Comment:
```suggestion
                        ty::PredicateKind::RegionOutlives(..) | ty::PredicateKind::ConstEvaluatable(..) => ControlFlow::CONTINUE
```
As I understand, these predicates just duplicate the constants otherwise visited by `visit_const`, and exist only to keep all such constants in a single list.
If that's not true, please make a test where this change makes a difference.

Reply:
i think so,

I expect this to change once explicit const evaluatable predicates are added though
as we won't visit them using `visit_const` here afaict

Reply:
Should I just leave this change out for now then?

Reply:
> Should I just leave this change out for now then, and just leave a FIXME(const_evaluatable_checked)?

I think you don't have to do anything in this PR for this comment.
We will have to add test once there is an actually known, and intended, difference in behavior, which is currently not the case.

Reply:
>I expect this to change once explicit const evaluatable predicates are added though

I see, I wasn't aware of the plans to introduce explicit const evaluatable predicates.

I'm surprised that the implicit const evaluatable predicates get into `explicit_predicates_of` though (which is the predicate set used by privacy).
Until the explicit predicates are implemented I'd prefer to see a panic here.

Reply:
Could you deduplicate the code between this place and `visit_const` if you keep it?

Reply:
> get into explicit_predicates_of though (which is the predicate set used by privacy).

yeah, that's just wrong rn :laughing: there's a reason this is still in `INCOMPLETE_FEAUTURES` ^^

Reply:
I've kept it & deduplicated it, but left this unresolved in case there's more I can add

Reply:
Looks like it's only partially deduplicated, why can't the `if let Ok(Some(ct)) = AbstractConst::new(tcx, defs, substs)` part be deduplicated as well?
(Also it's better to rename `walk_node_recur` to something meaningful, because with generic name like this it's entirely unclear what the function does unless you read its code.)

Reply:
Er, I can't put it near `fn visit_const` which is in a trait impl block, and I also can't move that part into the function because inside of `visit_const` it's `if let Ok(Some(ct)) = AbstractConst::from_const(tcx, c: &'tcx Const)`, but I can move one more line in